dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/ServiceBean.java

```diff
@@ -308,4 +253,89 @@ protected Class getServiceClass(T ref) {
+        if (applicationConfigMap != null && applicationConfigMap.size() > 0) {
+            ApplicationConfig applicationConfig = null;
+            for (ApplicationConfig config : applicationConfigMap.values()) {
+                if (config.isDefault() == null || config.isDefault()) {
+                    if (applicationConfig != null) {
+                        throw new IllegalStateException("Duplicate application configs: " + applicationConfig + " and " + config);
+                    }
+                    applicationConfig = config;
+                }
+            }
+            if (applicationConfig != null) {
+                setApplication(applicationConfig);
+            }
+        }
+    }
+
+    private void setupProvider(Map<String, ProviderConfig> providerConfigMap) {
+        Map<String, ProtocolConfig> protocolConfigMap = applicationContext == null ? null : BeanFactoryUtils.beansOfTypeIncludingAncestors(applicationContext, ProtocolConfig.class, false, false);
+        if ((protocolConfigMap == null || protocolConfigMap.size() == 0)
+                && providerConfigMap.size() > 1) { // backward compatibility
+            List<ProviderConfig> providerConfigs = new ArrayList<ProviderConfig>();
+            for (ProviderConfig config : providerConfigMap.values()) {
+                if (config.isDefault() != null && config.isDefault()) {
+                    providerConfigs.add(config);
+                }
+            }
+            if (!providerConfigs.isEmpty()) {
+                setProviders(providerConfigs);
+            }
+        } else {
+            ProviderConfig providerConfig = null;
+            for (ProviderConfig config : providerConfigMap.values()) {
+                if (config.isDefault() == null || config.isDefault()) {
+                    if (providerConfig != null) {
+                        throw new IllegalStateException("Duplicate provider configs: " + providerConfig + " and " + config);
+                    }
+                    providerConfig = config;
+                }
+            }
+            if (providerConfig != null) {
+                setProvider(providerConfig);
+            }
+        }
+    }
+
+    private boolean hasApplicationRegistry() {
+        return getApplication() != null && getApplication().getRegistries() != null && !getApplication().getRegistries().isEmpty();
+    }
+
+    private boolean hasProviderRegistry() {
+        return getProvider() != null && getProvider().getRegistries() != null && !getProvider().getRegistries().isEmpty();
+    }
+
+    private boolean hasRegistry() {
+        return getRegistries() != null && !getRegistries().isEmpty();
+    }
+
+    private boolean hasProviderProtocol() {
+        return getProvider() != null && getProvider().getProtocols() != null && getProvider().getProtocols().isEmpty();

```

Comment:
should be `!getProvider().getProtocols().isEmpty()`

Reply:
Thanks for catching this. I will add a UT for same too, so that in future it can catched .

Reply:
Great! I was thinking of adding unit test.